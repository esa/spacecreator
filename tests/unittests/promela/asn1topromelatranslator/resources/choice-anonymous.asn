CHOICES DEFINITIONS ::= BEGIN

SimpleChoiceWithAnonymousTypes ::= CHOICE 
{
    alt1 INTEGER(0..5),
    alt2 INTEGER(-1..1)
}

END