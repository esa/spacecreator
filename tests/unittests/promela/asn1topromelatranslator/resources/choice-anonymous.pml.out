#define SimpleChoiceWithAnonymousTypes_alt1_PRESENT 1
#define SimpleChoiceWithAnonymousTypes_alt2_PRESENT 2
inline SimpleChoiceWithAnonymousTypes_alt1_generate_value(value)
{
    if
    ::  true;
        value = 0;
    ::  true;
        value = 1;
    ::  true;
        value = 2;
    ::  true;
        value = 3;
    ::  true;
        value = 4;
    ::  true;
        value = 5;
   fi;
}
inline SimpleChoiceWithAnonymousTypes_alt2_generate_value(value)
{
    if
    ::  true;
        value = -1;
    ::  true;
        value = 0;
    ::  true;
        value = 1;
    fi;
}
inline SimpleChoiceWithAnonymousTypes_generate_value(value)
{
    if
    ::  true;
        SimpleChoiceWithAnonymousTypes_alt1_generate_value(value.alt1);
        value.selection = SimpleChoiceWithAnonymousTypes_alt1_PRESENT;
    ::  true;
        SimpleChoiceWithAnonymousTypes_alt2_generate_value(value.alt2);
        value.selection = SimpleChoiceWithAnonymousTypes_alt2_PRESENT;
    fi;
}
